mmaction/datasets/pipelines/loading.py

```diff
@@ -910,6 +910,122 @@ def __call__(self, results):
 
 
 @PIPELINES.register_module()
+class AudioDecodeInit(object):
+    """Using librosa to initialize the audio reader.
+
+    Args:
+        io_backend (str): io backend where frames are store.
+            Default: 'disk'.
+        sample_rate (int): Audio sampling times per second. Default: 16000.
+
+    Required keys are "audio_path", added or modified keys are "length",
+    "sample_rate", "audios".
+    """
+
+    def __init__(self, io_backend='disk', sample_rate=16000, **kwargs):
+        self.io_backend = io_backend
+        self.sample_rate = sample_rate
+
+        self.kwargs = kwargs
+        self.file_client = None
+
+    def __call__(self, results):
+        """Perform the librosa initialization.
+
+        Args:
+            results (dict): The resulting dict to be modified and passed
+                to the next transform in pipeline.
+        """
+        try:
+            import librosa
+        except ImportError:
+            raise ImportError('Please install librosa first.')
+
+        if self.file_client is None:
+            self.file_client = FileClient(self.io_backend, **self.kwargs)
+        try:
+            file_obj = io.BytesIO(self.file_client.get(results['audio_path']))
+            y, sr = librosa.load(file_obj, sr=self.sample_rate)
+        except BaseException:
+            # Generate a ramdom dummy 10s input
+            y = np.random.randn(10.0 * self.sample_rate, )
+            sr = self.sample_rate
+
+        results['length'] = y.shape[0]
+        results['sample_rate'] = sr
+        results['audios'] = y
+        return results
+
+
+@PIPELINES.register_module()
+class LoadAudioFeature(object):
+    """Load offline extracted audio features.
+
+    Required keys are "audio_path", added or modified keys are "length",
+    audios".
+    """
+
+    def __init__(self):
+        pass
+
+    def __call__(self, results):
+        """Perform the numpy loading.
+
+        Args:
+            results (dict): The resulting dict to be modified and passed
+                to the next transform in pipeline.
+        """
+        try:
+            feature_map = np.load(results['audio_path'])
+        except BaseException:
+            # Generate a ramdom dummy 10s input
+            # Some videos do not have audio stream
+            feature_map = np.random.randn(640, 80).astype(np.float32)
+        results['length'] = feature_map.shape[0]
+        results['audios'] = feature_map
+        return results
+
+
+@PIPELINES.register_module()
+class AudioDecode(object):
+    """Sample the audio w.r.t. the frames selected.
+
+    Required keys are "frame_inds", "num_clips", "total_frames", "length",
+    added or modified keys are "audios", "audios_shape".
+    """
+
+    def __init__(self):
+        np.warnings.filterwarnings(
+            'ignore', category=np.VisibleDeprecationWarning)
```

Comment:
what is this filter used for?

Reply:
Here,
```python
results['audios'] = np.array(resampled_clips)
```
Turning a list into numpy.ndarray will raise a warning due to latest numpy deprecates the usage of directly dtype transformation.

Reply:
so should concat or stack be used here?

Reply:
True, will fix it. Many other lines also use this operation though. Such as  https://github.com/open-mmlab/mmaction2/blob/bd25d00057e0a0ab36be92691bb43d0d95ab1bfd/mmaction/datasets/pipelines/loading.py#L315. https://github.com/open-mmlab/mmaction2/blob/bd25d00057e0a0ab36be92691bb43d0d95ab1bfd/mmaction/datasets/pipelines/loading.py#L831

Reply:
if they also raise warnings, then they need to be fixed. Can do this in a separate pr.

Reply:
Ah, my bad. This warning only occurs when the list to be transform into numpy is 'ragged'. I will revise this class to avoid it.